dashboard/app/models/ability.rb

```diff
@@ -344,6 +344,7 @@ def initialize(user)
         Lesson,
         ProgrammingEnvironment,
         ProgrammingExpression,
+        ReferenceGuide,
         CourseOffering,
         UnitGroup,
```

Comment:
Thank you for adding this update to ability.rb. This will make it so only levelbuilders can edit ReferenceGuides which is great. I think we also want to lock down ReferenceGuides to only be able to be viewed by instructor and participant audiences. I think we will need something like:

```
    can [:read], ReferenceGuide do |guide|
      course_or_unit = guide.course_version.content_root
      can?(:read, course_or_unit)
    end
```

I would add this near the line for Lesson `    can [:read, :show_by_id, :student_lesson_plan], Lesson do |lesson|`

Reply:
Will do! For my own understanding, what this is doing is checking the content root (UnitGroup or Script) and using the same permissions set for that. Is that right?

Reply:
Yup - Its checking if the user can view the UnitGroup or Script that associated with the course version. More about the library can be found here: https://github.com/CanCanCommunity/cancancan/blob/develop/docs/README.md

Reply:
Can you also add a test for this in ability.rb?